test/TimeZones.jl

```diff
@@ -1,5 +1,5 @@
 # Deserialization can cause us to have two immutables that are not using the same memory
-@test TimeZone("Europe/Warsaw") === TimeZone("Europe/Warsaw")
+@test tz"Europe/Warsaw" === "Europe/Warsaw"
```

Comment:
I'd like to keep the original test too since it's directly testing for what is mentioned in the comment.